ocaml-lsp-server/src/merlin_config.ml

```diff
@@ -389,9 +391,22 @@ let config (t : t) : Mconfig.t Fiber.t =
           let+ () = destroy t in
           use_entry entry
     in
+    let config_from_dot_merlin =
+      if !should_read_dot_merlin then
+        Some (Mconfig.get_external_config t.path t.initial)
+      else None
+    in
     let+ dot, failures = get_config entry.process ~workdir:ctx.workdir t.path in
-    let merlin = Config.merge dot t.initial.merlin failures config_path in
-    Mconfig.normalize { t.initial with merlin }
+
+    match config_from_dot_merlin with
```

Comment:
I think my original suggestion avoided running `Mconfig.get_external_config` where it wasn't unnecessary. As in the clause where `dot <> Config.empty`.

Reply:
My bad, I understand what you mean now.

Reply:
Just pushed a commit implementing your suggestion.